### DISCUSSION TITLE
How to avoid routing car to alleys?

### DISCUSSION THREAD

**Discussion by nvcnvn** [valhalla/valhalla#4246] [Q&A] [answered]
I guess one option is modify this factor https://github.com/valhalla/valhalla/blob/753c8e706e2e1850d2c1ffffdae2d7a4d387a69d/src/sif/autocost.cc#L54 

Do we have any other option?

**Comment by nilsnolde** [comment, marked as answer]
Jep, that's the only way. Would you need other ways of avoiding alleys? You can set it up to 10k in the request, so that'd add almost 3 hours of penalty. By default it's not affecting the cost at all. I think that's all fine.

**Reply by nilsnolde** [reply]
You might no be aware of all the request options maybe? https://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/

CTRL+F `alley_factor`

**Reply by nilsnolde** [reply]
just realizing it's falsely not listed in the `auto` request options, but it's also respected there (as you already found out)

**Reply by nvcnvn** [reply]
OK thank you for replying. I will go with the request option.
At first I think some how the `auto` costing model don't support alley factor but now yes then it should be a straightforward option.
